### PR TITLE
MI-1109 Added modal component and actions to set custom status

### PR DESCRIPTION
**Pull request by jatinkksharma**

### CODE REVIEW

File: actions/views/custom_status.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+export function updateUserCustomStatus(newCustomStatus: CustomStatus) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const userProps = Object.assign({}, getCurrentUser(getState()).props);
+        const recentCustomStatuses = userProps.recentCustomStatuses ? JSON.parse(userProps.recentCustomStatuses) : [];
+        const updatedRecentCustomStatuses = recentCustomStatuses.filter((status: CustomStatus) => status.text !== newCustomStatus.text);
+        updatedRecentCustomStatuses.unshift(newCustomStatus);
+        if (updatedRecentCustomStatuses.length > 5) {
+            updatedRecentCustomStatuses.pop();
+        }
```
Comment (chetanyakan) [inline actions/views/custom_status.ts]:
```suggestion
        const updatedRecentCustomStatuses = [
            newCustomStatus,
            ...recentCustomStatuses
                .filter((status: CustomStatus) => status.text !== newCustomStatus.text)
                .slice(0,4),
        ];
```

File: actions/views/custom_status.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {updateMe} from 'mattermost-redux/actions/users';
+import {DispatchFunc, GetStateFunc} from 'mattermost-redux/types/actions';
+
+import {CustomStatus} from 'types/store/custom_status';
+
+export function updateUserCustomStatus(newCustomStatus: CustomStatus) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const userProps = Object.assign({}, getCurrentUser(getState()).props);
```
Comment (chetanyakan) [inline actions/views/custom_status.ts]: Why do we need to use `Object.assign` here?

File: actions/views/custom_status.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        const recentCustomStatuses = userProps.recentCustomStatuses ? JSON.parse(userProps.recentCustomStatuses) : [];
+        const updatedRecentCustomStatuses = recentCustomStatuses.filter((status: CustomStatus) => status.text !== newCustomStatus.text);
+        updatedRecentCustomStatuses.unshift(newCustomStatus);
+        if (updatedRecentCustomStatuses.length > 5) {
+            updatedRecentCustomStatuses.pop();
+        }
+        userProps.customStatus = JSON.stringify(newCustomStatus);
+        userProps.recentCustomStatuses = JSON.stringify(updatedRecentCustomStatuses);
+        const user = Object.assign({}, getCurrentUser(getState()), {props: userProps});
```
Comment (chetanyakan) [inline actions/views/custom_status.ts]: Again, why `Object.assign`?

Also, you can use `getCurrentUser(getState())` just once.

**Review by chetanyakan — CHANGES_REQUESTED** [review]

**Review by chetanyakan — APPROVED** [review]

File: actions/views/custom_status.ts (hunk 1 above)
Comment (jatinkksharma) [reply actions/views/custom_status.ts]: Done

File: actions/views/custom_status.ts (hunk 2 above)
Comment (jatinkksharma) [reply actions/views/custom_status.ts]: I have updated this as discussed

**Pull request merged**